tests/test_dirty_local.py

```diff
@@ -0,0 +1,107 @@
+import os
+from pathlib import Path
+from shutil import copytree
+
+import pytest
+from plumbum import local
+from plumbum.cmd import git
+
+import copier
+from copier.errors import DirtyLocalWarning
+from copier.main import run_copy, run_update
+
+from .helpers import DATA, PROJECT_TEMPLATE, build_file_tree, filecmp
+
+
+def test_copy(tmp_path_factory):
+    src, dst = map(tmp_path_factory.mktemp, ("src", "dst"))
+
+    copytree(PROJECT_TEMPLATE, src, dirs_exist_ok=True)
+
+    with local.cwd(src):
+        git("init")
+
+    with pytest.warns(DirtyLocalWarning):
+        copier.copy(str(src), str(dst), data=DATA, quiet=True)
+
+    generated = (dst / "pyproject.toml").read_text()
+    control = (Path(__file__).parent / "reference_files/pyproject.toml").read_text()
+    assert generated == control
+
+    # assert template still dirty
+    assert bool(git("status", "--porcelain").strip())
+
+
+# Will fail due to lingering deleted file until #461 is fixed
```

Comment:
I use the keyword HACK to track... well, hacks 😄 

Also TODO to make clear what to do after the HACK is solved.

```suggestion
# Will fail due to lingering deleted file.
# HACK https://github.com/copier-org/copier/issues/461
# TODO Remove xfail decorator when fixed.
```